File: 01_basics/01_08_all_divisors/alldivisors.cpp
#include <bits/stdc++.h>

using namespace std;

void printAllDivisors(int n)
{
    cout << "The Divisors of " << n << " are: ";
    for (size_t i = 1; i < sqrt(n) + 1; i++)
    {
        if (n % i == 0)
        {
            cout << i << " ";
            if (i != n / i)
            {
                cout << n / i << " ";
            }
        }
    }
}

int main()
{
    int num = 36;

    printAllDivisors(num);
}

File: 01_basics/01_08_all_divisors/alldivisors.py
def print_all_divisors(n):
    print(f"All divisors of {n} are: ", end="")
    for i in range(1, int(n**0.5) + 1):
        if n % i == 0:
            print(f"{i}", end=" ")
            if i != n//i:
                print(f"{n//i}", end=" ")
            
num = 36
print_all_divisors(num)